common/metrics/otel_event_handler.go

```diff
@@ -90,18 +86,7 @@ func (m *OtelMetricHandler) Event(ctx context.Context, e *event.Event) context.C
 	return ctx
 }
 
-func (m *OtelMetricHandler) getRecordFunc(em event.Metric) recordFunc {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	if f, ok := m.recordFuncs[em]; ok {
```

Comment:
so event.Metric is unbounded? 

And instead of caching recordFuns, we create a new one every time? Will it impact latency or gc?